<a href="https://colab.research.google.com/github/YaraMansour1985/Python-Functions/blob/main/slicing_Notebook.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Insurance Data Starter Notebook

In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


Student Name:yara mansour  

## Table of Contents


This notebook will be used for 4 practice assignments:
- [📝 Loading Data (Practice)](#loading)
- [📝 Slicing (Practice)](#slicing)
- [📝 Filtering (Practice)](#filtering)


Click on the assignment names above to jump to the corresponding headers on Google Colab.  



___




Note that:
  - The first assignment (Loading Data) must be completed before the other three assignments may be completed.

  - We recommend you do each of the remaining sections as you encounter them on the Learning Platform.


<a name="loading"> </a>
# 📝 Loading Data (Practice)

[Insurance Data](https://drive.google.com/file/d/1-7mcBguuzAV7JWVu2XCpgAO-fO6X2cl0/view?usp=sharing)

1) Open the link and save the shared file to your Drive in the week 2 data folder.

2) Mount your drive

3) Find and copy the file path of the insurance_mod.csv file

In [ ]:
# Save the copied file path as a variable
# Surround the path with quotes
fname=("/content/drive/MyDrive/AXSOSACADEMY/01-Fundamentals/Week02/Data/insurance_mod.csv")



4) Load in the file as a Pandas dataframe named df_insurance


In [ ]:
# Include needed import(s)
import pandas as pd

# Save the copied file path as a variable
# Surround the path with quotes
fname=("/content/drive/MyDrive/AXSOSACADEMY/01-Fundamentals/Week02/Data/insurance_mod.csv")

# Load in the file as a Pandas dataframe named df_insurance
df_insurance=pd.read_csv(fname)

In [ ]:
# Include needed import(s)
import pandas as pd

# Load in the file as a Pandas dataframe named df_insurance
df_insurance=pd.read_csv(fname)


5) Preview the first 5 rows with .head()



In [ ]:
# Preview the first 5 rows with .head()
df_insurance.head()

age     sex     bmi  children  smoker     region  charges
0   19  female  27.900         0       1  southwest  16885.0
1   18    male  33.770         1       0  southeast   1726.0
2   28    male  33.000         3       0  southeast   4449.0
3   33    male  22.705         0       0  northwest  21984.0
4   32    male  28.880         0       0  northwest   3867.0

6) Preview the last 5 rows with .tail()

In [ ]:
# Preview the last 5 rows with .tail()
df_insurance.tail()


age     sex    bmi  children  smoker     region  charges
1333   50    male  30.97         3       0  northwest  10601.0
1334   18  female  31.92         0       0  northeast   2206.0
1335   18  female  36.85         0       0  southeast   1630.0
1336   21  female  25.80         0       0  southwest   2008.0
1337   61  female  29.07         0       1  northwest  29141.0

7) Obtain summary information with .info()

In [ ]:
# Obtain summary information with .info()
df_insurance.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1338 entries, 0 to 1337
Data columns (total 7 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   age       1338 non-null   int64  
 1   sex       1338 non-null   object 
 2   bmi       1338 non-null   float64
 3   children  1338 non-null   int64  
 4   smoker    1338 non-null   int64  
 5   region    1338 non-null   object 
 6   charges   1338 non-null   float64
dtypes: float64(2), int64(3), object(2)
memory usage: 73.3+ KB


8) Use the output of .info() to answer the following in a text cell below:




a) How many rows are there?

- Answer:1338

b) How many columns are there?

- Answer:7

## You have completed the "Loading Data" Practice Assignment.



- 🛑 Stop here.

- 💾 Save and submit this notebook now.

- 📚 Continue working through the lessons.

- ⏎ Return to your copy of this notebook for the next practice assignment.

___


<a name="slicing"> </a>
# 📝 Slicing (Practice)


1) Slice the row for the customer at index 50 as a Series

In [ ]:
# Use single brackets so the output is a Pandas Series
import pandas as pd
fname=("/content/drive/MyDrive/AXSOSACADEMY/01-Fundamentals/Week02/Data/insurance_mod.csv")
df_insurance=pd.read_csv(fname)
df_insurance.iloc[50]

age                18
sex            female
bmi            35.625
children            0
smoker              0
region      northeast
charges        2211.0
Name: 50, dtype: object

In [ ]:


# Load in the file as a Pandas dataframe named df_insurance
df_insurance=pd.read_csv(fname)


2) Slice the row for the customer at index 50 as a DataFrame

In [ ]:
# Use double brackets so the output is a Pandas Dataframe
df_insurance.iloc[[50]]

age     sex     bmi  children  smoker     region  charges
50   18  female  35.625         0       0  northeast   2211.0

3) Find out if the customer at index 20 was a male or female (by slicing index 20 and column "Sex")

In [ ]:
# find out if the customer at index 20 was a male or female by slicing index 20 and column "Sex"
df_insurance.iloc[20, df_insurance.columns.get_loc("sex")]

'female'

4) Slice the entire smoker column using square brackets


In [ ]:
# Slice the entire smoker column using square brackets
df_insurance["smoker"]


0       1
1       0
2       0
3       0
4       0
       ..
1333    0
1334    0
1335    0
1336    0
1337    1
Name: smoker, Length: 1338, dtype: int64

5) Slice the entire smoker column using .loc


In [ ]:
# Slice the entire smoker column using .loc
df_insurance.loc[:,"smoker"]


0       1
1       0
2       0
3       0
4       0
       ..
1333    0
1334    0
1335    0
1336    0
1337    1
Name: smoker, Length: 1338, dtype: int64

6) Slice the entire smoker column using either method, but this time save it as a variable called "smoker"


In [ ]:
# Slice the entire smoker column using either method, but this time save it as a variable called "smoker"
smoker=df_insurance.loc[:,"smoker"]



- Calculate the sum of the smoker variable to see how many of the customers smoke


In [ ]:
# To see how many of the customers smoke, get the sum
df_insurance["smoker"].sum()

np.int64(274)


- Calculate the mean of the smoker variable using .mean() to see the ratio of smokers in the dataset


In [ ]:
# To get the ratio use .mean()
df_insurance["smoker"].mean()

np.float64(0.20478325859491778)

## You have completed the "Slicing Data" Practice Assignment.



- 🛑 Stop here.

- 💾 Save and submit this notebook now.

- 📚 Continue working through the lessons.

- ⏎ Return to your copy of this notebook for the next practice assignment.

___


<a name="filtering"> </a>
# 📝 Filtering (Practice)

1) Create a filter variable called under_30_filter for rows where the age is less than 30



In [ ]:
# Create a filter called under_30_filter for the rows where the age is less than 30
df_insurance=pd.read_csv(fname)
fname=("/content/drive/MyDrive/AXSOSACADEMY/01-Fundamentals/Week02/Data/insurance_mod.csv")
df_insurance["age<30"]=df_insurance["age"]<30
under_30_filter=df_insurance["age<30"]

In [ ]:
# View the under_30_filter to see the boolean index
under_30_filter


0        True
1        True
2        True
3       False
4       False
        ...  
1333    False
1334     True
1335     True
1336     True
1337    False
Name: age<30, Length: 1338, dtype: bool

In [ ]:
# Get the sum of the true values for the under_30_filter
# This is the number of customers who are under the age of 30
df_insurance["age<30"].sum()


np.int64(417)

In [ ]:
# # Use the filter to view only rows with customers who are under 30
df_insurance[under_30_filter]


age     sex     bmi  children  smoker     region  charges  age<30
0      19  female  27.900         0       1  southwest  16885.0    True
1      18    male  33.770         1       0  southeast   1726.0    True
2      28    male  33.000         3       0  southeast   4449.0    True
10     25    male  26.220         0       0  northeast   2721.0    True
12     23    male  34.400         0       0  southwest   1827.0    True
...   ...     ...     ...       ...     ...        ...      ...     ...
1328   23  female  24.225         2       0  northeast  22396.0    True
1331   23  female  33.400         0       0  southwest  10796.0    True
1334   18  female  31.920         0       0  northeast   2206.0    True
1335   18  female  36.850         0       0  southeast   1630.0    True
1336   21  female  25.800         0       0  southwest   2008.0    True

[417 rows x 8 columns]

2) Create a filter for males

In [ ]:
# Create a filter for males called male_filter
df_insurance=pd.read_csv(fname)
fname=("/content/drive/MyDrive/AXSOSACADEMY/01-Fundamentals/Week02/Data/insurance_mod.csv")
df_insurance["male_filter"]=df_insurance["sex"]=="male"
male_filter = df_insurance["male_filter"]

In [ ]:
# View the filter to see the boolean index
df_insurance=pd.read_csv(fname)
fname=("/content/drive/MyDrive/AXSOSACADEMY/01-Fundamentals/Week02/Data/insurance_mod.csv")
df_insurance["male_filter"]=df_insurance["sex"]=="male"
male_filter = df_insurance["male_filter"]
male_filter

0       False
1        True
2        True
3        True
4        True
        ...  
1333     True
1334    False
1335    False
1336    False
1337    False
Name: male_filter, Length: 1338, dtype: bool

In [ ]:
# Obtain the sum of True values in the male_filter
# This is the number of customers who are male
df_insurance["male_filter"].sum()



np.int64(676)

In [ ]:
# Use the filter to view only rows with customers who are male
df_insurance[male_filter]


age   sex     bmi  children  smoker     region  charges  male_filter
1      18  male  33.770         1       0  southeast   1726.0         True
2      28  male  33.000         3       0  southeast   4449.0         True
3      33  male  22.705         0       0  northwest  21984.0         True
4      32  male  28.880         0       0  northwest   3867.0         True
8      37  male  29.830         2       0  northeast   6406.0         True
...   ...   ...     ...       ...     ...        ...      ...          ...
1324   31  male  25.935         1       0  northwest   4240.0         True
1325   61  male  33.535         0       0  northeast  13143.0         True
1327   51  male  30.030         1       0  southeast   9378.0         True
1329   52  male  38.600         2       0  southwest  10325.0         True
1333   50  male  30.970         3       0  northwest  10601.0         True

[676 rows x 8 columns]

3) Combine the under_30_filter and the male_filter to create a single filter called under_30_male_filter

In [ ]:
# Combine the under_30_filter and the male_filter to create a single filter called under_30_male_filter
under_30_male_filter=under_30_filter & male_filter


In [ ]:
# Display the boolean index of the under_30_male_filter
under_30_male_filter


0       False
1        True
2        True
3       False
4       False
        ...  
1333    False
1334    False
1335    False
1336    False
1337    False
Length: 1338, dtype: bool

In [ ]:
# Get the sum of the under_30_male_filter True values
# This is the number of customers who are both under 30 and male
df_insurance["male_filter"].sum()


np.int64(676)

In [ ]:
# Use the filter to view only rows with customers who are both under 30 and male
df_insurance[under_30_male_filter]


age   sex     bmi  children  smoker     region  charges  male_filter
1      18  male  33.770         1       0  southeast   1726.0         True
2      28  male  33.000         3       0  southeast   4449.0         True
10     25  male  26.220         0       0  northeast   2721.0         True
12     23  male  34.400         0       0  southwest   1827.0         True
14     27  male  42.130         0       1  southeast  39612.0         True
...   ...   ...     ...       ...     ...        ...      ...          ...
1292   21  male  23.210         0       0  southeast   1515.0         True
1295   20  male  22.000         1       0  southwest   1965.0         True
1296   18  male  26.125         0       0  northeast   1709.0         True
1315   18  male  28.310         1       0  northeast  11272.0         True
1317   18  male  53.130         0       0  southeast   1163.0         True

[216 rows x 8 columns]

4) Create a filter for regions that contain south using .str.contains



In [ ]:
# Create a filter for regions that contain south using .str.contains
df_insurance=pd.read_csv(fname)
fname=("/content/drive/MyDrive/AXSOSACADEMY/01-Fundamentals/Week02/Data/insurance_mod.csv")
df_insurance["south_filter"]=df_insurance["region"].str.contains("south")
south_filter=df_insurance["south_filter"]


In [ ]:
# Use the filter to view only rows with regions containing "south"
df_insurance[south_filter]


age     sex    bmi  children  smoker     region  charges  south_filter
0      19  female  27.90         0       1  southwest  16885.0          True
1      18    male  33.77         1       0  southeast   1726.0          True
2      28    male  33.00         3       0  southeast   4449.0          True
5      31  female  25.74         0       0  southeast   3757.0          True
6      46  female  33.44         1       0  southeast   8241.0          True
...   ...     ...    ...       ...     ...        ...      ...           ...
1330   57  female  25.74         2       0  southeast  12629.0          True
1331   23  female  33.40         0       0  southwest  10796.0          True
1332   52  female  44.70         3       0  southwest  11412.0          True
1335   18  female  36.85         0       0  southeast   1630.0          True
1336   21  female  25.80         0       0  southwest   2008.0          True

[689 rows x 8 columns]

## You have completed the "Filtering Data" Practice Assignment.



- 🛑 Stop here.

- 💾 Save and submit this notebook now.

- 📚 Continue working through the lessons.

- ⏎ Return to your copy of this notebook for the next practice assignment.

___
